src/kOS.Safe/Execution/CPU.cs

```diff
@@ -1121,77 +1111,96 @@ private void PostUpdateBindings()
             }
         }
 
-        private void ProcessWait()
-        {
-            if (currentStatus == Status.Waiting)
-            {
-                if (currentTime >= timeWaitUntil)
-                {
-                    EndWait();
-                }
-            }
-        }
-
-        private void ProcessTriggers(bool doProfiling)
+        private void ProcessTriggers()
         {
-            if (currentContext.Triggers.Count <= 0) return;
+            if (currentContext.ActiveTriggerCount() <= 0) return;
             int oldCount = currentContext.Program.Count;
 
             int currentInstructionPointer = currentContext.InstructionPointer;
-            var triggerList = new List<int>(currentContext.Triggers);
-
-            foreach (int triggerPointer in triggerList)
+            var triggersToBeExecuted = new List<int>();
+            
+            // To ensure triggers execute in the same order in which they
+            // got added (thus ensuring the system favors trying the least
+            // recently added trigger first in a more fair round-robin way),
+            // the nested function calls get built in stack order, so that
+            // they execute in normal order.  Thus the backward iteration
+            // order used in the loop below:
+            for (int index = currentContext.ActiveTriggerCount() - 1 ; index >= 0 ; --index)
             {
-                try
```

Comment:
I hadn't realized that it didn't beforehand.  I swear I had cases where the exception in a trigger broke main code, rather than re-firing again and again.